### PR TITLE
Update AdditionalPayments task to use default gateway suggestion

### LINKED ISSUE #50543: Investigate other synchronous calls to remote endpoints
### Issue Description

P2: pdibGW-3xH-p2

It was discovered that the Marketing task and [WooCommercePayments task](https://github.com/woocommerce/woocommerce/blob/db5eb6baa63ef2cb2a1220c0718019121543b9d9/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/WooCommercePayments.php#L182) had synchronous calls to Remote Specs API fetch. Marketing task was [fixed](https://github.com/woocommerce/woocommerce/pull/50479), but there could be more remote endpoint calls that could potentially break wp-admin.

### Acceptance criteria

1. Investigate and ensure no other calls to external endpoints are synchronous
2. Fix if found

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes #50543

Prevents synchronous Remote Specs fetch when calculating tasks' view & completion status.

Changes:

1. Adds a default gateway suggestion spec method
2. Updates AdditionalPayments task to use the new default method

### How to test the changes in this Pull Request:

**Scenario: No payment gateway is setup**

1. Select `United States` as your store country
2. Ensure WooPayments plugin is not installed
5. Go to `WooCommerce > Home`
6. Observe that `Set up additional payment options` task is **not** shown in the extended tasklist

**Scenario: WooPayments is setup**

1. Install and activate fake plugin [woocommerce-payments.zip](https://github.com/user-attachments/files/16611850/woocommerce-payments.zip)
1. Go to `WooCommerce > Home`
1. Observe that  `Set up additional payment options` task is shown
8. Click on `Set up additional payment options` task
9. You should see the `Additional payment options` payment lists as per [screenshot](https://github.com/user-attachments/assets/f8ccb327-db98-4f89-ad08-08f55a42194f)
1. Install and activate fake plugin [klarna-payments-for-woocommerce.zip](https://github.com/user-attachments/files/16611852/klarna-payments-for-woocommerce.zip)
14. Go to `WooCommerce > Home`
15. Observe that `Set up additional payment options` task is marked as completed.

**Scenario: "other category" payment is setup**

1. Deactivate  `woocommerce-payments`
4. Go to `WooCommerce > Home`
5. Observe that  `Set up additional payment options` task is not shown
1. Install and activate fake plugin [stripe.zip](https://github.com/user-attachments/files/16611853/stripe.zip)
8. Go to `WooCommerce > Home`
9. Observe that  `Set up additional payment options` task is shown
11. Activate `klarna-payments-for-woocommerce` plugin
12. Observe that `Set up additional payment options` task is marked as completed.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php [hunk 1]
```diff
@@ -61,6 +61,24 @@ public static function get_suggestions( array $specs = null ) {
 		return $specs_to_return;
 	}
 
+	/**
+	 * Gets the default suggestions.
+	 *
+	 * @return array
+	 */
+	public static function get_default_suggestions() {
+		/**
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php:77]: Old filter lint issue

**Review by chihsuan — COMMENTED** [review]
Code looks good to me! 👍 

~I didn't test scenario 3 because I think it needs to set up Stripe? I was able to fake the woo payments completion status by modifying the code to return `true` from [needs_setup](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L801) and `is_connected` and confirmed it showed the lists as the screenshot.~  Just realized that the zip files you provided contain the necessary files to test the scenarios. 😅 

I wonder if we change the wccom endpoint to return different additional payment options later, will users be able to view/complete the task as the expected?  I didn't test this scenario, but I think it's worth considering. 

Just left comments. I'll let others to approve this PR.  🙏

**Comment by ilyasfoo** [comment]
Thanks for the feedback, @chihsuan!

> I wonder if we change the wccom endpoint to return different additional payment options later, will users be able to view/complete the task as the expected? I didn't test this scenario, but I think it's worth considering.

Thanks a good point, it's possible for the "additional" category gateways will eventually fall out of sync from the defaults. I'm thinking instead of always getting defaults, we could do look for cache with fallback to default specs 🤔

**Comment by ilyasfoo** [comment]
@chihsuan Added `get_cached_or_default_suggestions` method here [9c7b5e3](https://github.com/woocommerce/woocommerce/pull/50674/commits/9c7b5e39eb3f54852347392775e82d482f7e172f) and tests in c13e5faba254aae968c8670b7f8d3030141a008c

File: plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php [hunk 2]
```diff
@@ -61,6 +61,29 @@ public static function get_suggestions( array $specs = null ) {
...
+	/**
+	 * Gets either cached or default suggestions.
+	 *
+	 * @return array
+	 */
+	public static function get_cached_or_default_suggestions() {
+		$specs = PaymentGatewaySuggestionsDataSourcePoller::get_instance()->get_cached_specs();
+
+		if ( ! is_array( $specs ) || 0 === count( $specs ) ) {
+			$specs = DefaultPaymentGateways::get_all();
```
Comment (rjchow) [inline plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php:75]: Noting that for users with tracking opted-out, they will always hit this path. I don't think there's anything wrong with that though

**Review by rjchow — COMMENTED** [review]
Hey @ilyasfoo nice work here! Your testing instruction works as expected, but I was also attempting to test that WC Installations with tracking enabled still receive the remote specs. Do you know how we might be able to assert that?

File: plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php (hunk 2 above)
Comment (ilyasfoo) [reply plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/Init.php:75]: Good catch, I think it should return defaults for consistent behaviour. Added the check in 7974d3d9e75e2fc2910467126bda2c5146a5a26b

**Comment by ilyasfoo** [comment]
> Your testing instruction works as expected, but I was also attempting to test that WC Installations with tracking enabled still receive the remote specs. Do you know how we might be able to assert that?

Good catch, @rjchow! To be accurate, it's the marketplace suggestions option, not tracking (I confirm that we still query remote specs with tracking disabled). I've added the test in 5914a88bcffe453c5fafb37b406e8323bc2acd79

**Review by chihsuan — APPROVED** [review]
> Added get_cached_or_default_suggestions method here [9c7b5e3](https://github.com/woocommerce/woocommerce/pull/50674/commits/9c7b5e39eb3f54852347392775e82d482f7e172f) and tests in https://github.com/woocommerce/woocommerce/commit/c13e5faba254aae968c8670b7f8d3030141a008c

Thanks for addressing this! @ilyasfoo Nice approach. LGTM and tested well. 👍

**Review by rjchow — APPROVED** [review]
Thanks for addressing the feedback!

**Pull request merged**